cob_omni_drive_controller/src/stuck_detector.cpp

```diff
@@ -0,0 +1,59 @@
+#include <std_srvs/Trigger.h>
+#include <cob_omni_drive_controller/WheelCommands.h>
+#include <ros/ros.h>
+
+ros::ServiceClient g_halt_client;
+std::vector<ros::Time> g_last_ok;
+double g_threshold;
+ros::Duration g_timeout;
+bool g_halt_requested;
+
+void commandsCallback(const cob_omni_drive_controller::WheelCommands::ConstPtr& msg){
+    g_last_ok.resize(msg->steer_target_error.size(), msg->header.stamp);
+
+    bool valid = true;
+    for(size_t i = 0; i < msg->steer_target_error.size(); ++i){
+
+        if(fabs(msg->steer_target_error[i]) >= g_threshold){
+            valid = false;
+            if( (msg->header.stamp - g_last_ok[i]) >= g_timeout  && !g_halt_requested) {
+                g_halt_requested = true;
+                std_srvs::Trigger srv;
+                ROS_ERROR_STREAM("Wheel " << i << " exceeded threshold for too long, halting..");
+                g_halt_client.call(srv);
```

Comment:
I can check the return code, but It doesn't matter.
I don't see any reason for the (ros_canopen) halt-service to fail, except for bus crashes etc.
In this case the base will stop anyway.

I could issue an error message, but it might confuse users more than it helps..